src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,715 @@
+            kv_json = {'value': kv.value}
+            res[kv.key] = kv_json
+    # import/export key, value, and tags (same level as key-value)
+    elif level == 'appservice':
+        for kv in keyvalues:
+            kv_json = {'value': kv.value}
+            if kv.tags:
+                for tag_k, tag_v in kv.tags.items():
+                    if tag_k == 'AppService:SlotSetting':
+                        kv_json[tag_k] = tag_v
+            res[kv.key] = kv_json
+    # import/export key, value, content-type, and tags (as a sub group)
+    elif level == 'appconfig':
+        for kv in keyvalues:
+            # value
+            kv_json = {'value': kv.value}
+            # tags
+            tag_json = {}
+            if kv.tags:
+                for tag_k, tag_v in kv.tags.items():
+                    tag_json[tag_k] = tag_v
+            kv_json['tags'] = tag_json
+            # content type
+            if kv.content_type:
+                kv_json['content type'] = kv.content_type
+            else:
+                kv_json['content type'] = ""
+            # key
+            res[kv.key] = kv_json
+    return res
+
+
+def __serialize_features_from_kv_list_to_comparable_json_object(keyvalues):
+    res = {}
+    for kv in keyvalues:
+        feature = map_keyvalue_to_featureflag(kv)
+        # state
+        feature_json = {'state': feature.state}
+        # conditions
+        feature_json['conditions'] = feature.conditions
+        # key
+        res[feature.key] = feature_json
+    return res
+
+
+def __serialize_feature_list_to_comparable_json_object(features):
+    res = {}
+    for feature in features:
+        # state
+        feature_json = {'state': feature.state}
+        # description
+        feature_json['description'] = feature.description
+        # conditions
+        feature_json['conditions'] = feature.conditions
+        # key
+        res[feature.key] = feature_json
+    return res
+
+
+def __serialize_kv_list_to_comparable_json_list(keyvalues):
+    res = []
+    for kv in keyvalues:
+        # value
+        kv_json = {'key': kv.key,
+                   'value': kv.value,
+                   'label': kv.label,
+                   'last modified': kv.last_modified,
+                   'content type': kv.content_type}
+        # tags
+        tag_json = {}
+        if kv.tags:
+            for tag_k, tag_v in kv.tags.items():
+                tag_json[tag_k] = tag_v
+        kv_json['tags'] = tag_json
+        res.append(kv_json)
+    return res
+
+
+def __print_features_preview(old_json, new_json):
+    logger.warning('\n---------------- Feature Flags Preview (Beta) -------------')
+    if not new_json:
+        logger.warning('\nSource configuration is empty. No changes will be made.')
+        return False
+
+    # perform diff operation
+    # to simplify output, add one shared key in src and dest configuration
+    new_json['@base'] = ''
+    old_json['@base'] = ''
+    differ = JsonDiffer(syntax='symmetric')
```

Comment:
>symmetric [](start = 32, length = 9)

why we use symmetric for feature flag but explicit for key values?

Reply:
explicit syntax doesn't work when comparing feature filters, probably due to whitespacing issue with the json formatting of filters. 

---
In reply to: [356899926](https://github.com/Azure/azure-cli/pull/11467#discussion_r356899926) [](ancestors = 356899926)